extensions/extjwt.md

```diff
@@ -12,71 +12,82 @@ copyrights:
-2. Improves security in that a misconfigured or vulnerable external service cannot impact the IRC network or user accounts
-3. Provides a common method to verify user status on a channel (joined, channel modes) while using well known libraries and methods that are available for many languages and frameworks (JWT)
+1. Makes it easier to deploy an external service as it does not need access to your IRC server.
+2. Improves security in that a misconfigured or vulnerable external service cannot impact the IRC network or user accounts.
+3. Provides a common method to verify user status on a channel (joined, channel modes) while using well known libraries and methods that are available for many languages and frameworks (JWT).
 
-### How the proof works
+### How the verification works
 
-The IRC network creates the proof by generating signed JWT tokens and sending them to the client. The client may use this token to open an external service with the token in its URL.
+The IRC network generates signed JWT tokens and sends them to the client. The client may use this token to open an external service with the token in its URL.
 
-The JWT token (https://jwt.io/) is an encoded JSON payload that is signed with a shared secret string between the IRC server and the external service. The JSON payload consists of known properties (claims) that include:
+The JWT token (https://jwt.io/) is a base64 encoded JSON payload that is signed with a secret string. The JSON payload consists of known claims that include:
 * `exp` `1529917513` Expiry time for this token.
 * `iss` `"irc.example.org"` The server name that generated this token.
 * `sub` `"somenick"` The nick of the user that generated this token.
+* `vfy` `"https://irc.example.org/extjwtverify/?t=%s"` The URL that can verify this token.
 * `account` `"somenick"` The account name of the user that generated this token. Empty if not logged in.
-* `net_modes` `["o"]` User modes the IRCd wishes to disclose. Eg, if the user is an operator.
+* `umodes` `["o"]` User modes the IRCd wishes to disclose. Eg, if the user is an operator.
 
-When an external service is opened with this token in its URL, the external service verifies that the token has not been tampered with using its pre-configured secret string and can then use the available claims to create any required user accounts and log the user in automatically.
+When an external service is opened with this token in its URL, the external service verifies that the token has not been tampered with by one of two methods:
+1. If the external service has a shared secret for this IRC network, it can now verify the token directly.
+2. The external service can make a HTTP GET request to the URL given in the `vfy` claim, replacing `%s` with the token string. The request MUST be responded to with a HTTP 200 status if the token has been verified, or a HTTP 403 status if the token is invalid.
```

Comment:
```suggestion
2. The external service can make a HTTP GET request to the URL given in the `vfy` claim, replacing `%s` with the token string. The request MUST be responded to with a HTTP 200 status if the token has been verified, or a HTTP 401 status if the token is invalid.
```

403 usually indicates that the token is valid, but doesn’t have access to the particular resource (AuthN vs AuthZ).

Secondly, can we relax the 200 response to something such as a “successful” response. If the API doesn’t have a response body then it might want to return a 204 which would be against this spec’s MUST requirement.